### PR TITLE
Add function to test if a world point is contained in fwd transform's domain

### PR DESCRIPTION
**Pull request by mcara**
Added a function that tests if input world coordinate is contained within forward transformation's image and that it maps to the domain of definition of the forward transformation. In practical terms, this function tests that input world coordinate can be converted to input frame and that it is within the forward transformation's ``bounding_box`` when defined.

### CODE REVIEW

File: gwcs/wcs.py [hunk 1]
```diff
@@ -334,6 +334,61 @@ def _invert(self, *args, **kwargs):
...
+        args : float, array like, `~astropy.coordinates.SkyCoord` or
+            `~astropy.units.Unit` coordinates to be inverted
+
+        kwargs : dict
+            keyword arguments to be passed either to ``backward_transform``
+            (when defined) or to the iterative invert method.
+
+        Returns
+        -------
+        result : bool
```
Comment (perrygreenfield) [inline gwcs/wcs.py:357]: boolean or boolean array?

Comment (mcara) [reply gwcs/wcs.py:357]: I am fixing this. It is both (depending on the input and number of axes in the input frame)

**Review by perrygreenfield — APPROVED** [review]

File: gwcs/wcs.py [hunk 2]
```diff
@@ -334,6 +334,63 @@ def _invert(self, *args, **kwargs):
...
+        the domain of definition of the forward transformation.
+        In practical terms, this function tests
+        that input world coordinate(s) can be converted to input frame and that
+        it is within the forward transformation's ``bounding_box`` when
+        defined.
+
+        Parameters
+        ----------
+        args : float, array like, `~astropy.coordinates.SkyCoord` or
+            `~astropy.units.Unit` coordinates to be inverted
```
Comment (nden) [inline gwcs/wcs.py:350]: Did you mean `~astropy.units.Quantity`, not `Unit` here? 
And does this work for mixed spatial and spectral?

Comment (mcara) [reply gwcs/wcs.py:350]: @nden This line(s) were "borrowed" from the docstring for `invert()`. Should that one fixed as well?

Comment (mcara) [reply gwcs/wcs.py:350]: @nden Tested on a MIRI image with mixed spatial and spectral WCS. Seems to work as expected.

**Comment by nden** [comment]
@mcara Please rebase

**Review by nden — APPROVED** [review]

**Pull request merged**